caml_z.c

```diff
@@ -741,6 +742,25 @@ CAMLprim value ml_z_to_nativeint(value v)
   return caml_copy_nativeint(x);
 }
 
+CAMLprim value ml_z_to_nativeint_unsigned(value v)
+{
+  uintnat r;
+  Z_DECL(v);
+  Z_MARK_OP;
+  Z_CHECK(v);
+  if (Is_long(v)) {
+    intnat x = Long_val(v);
+    if (x < 0) ml_z_raise_overflow();
+    return caml_copy_nativeint((uintnat)x);
+  }
+  Z_MARK_SLOW;
+  Z_ARG(v);
+  if (!size_v) return r = 0;
```

Comment:
`return` should not be there.
```suggestion
  if (!size_v) r = 0;
```